bundles/org.openhab.binding.easee/src/main/java/org/openhab/binding/easee/internal/connector/WebInterface.java

```diff
@@ -222,7 +222,8 @@ private synchronized void authenticate() {
         private synchronized void refreshAccessToken() {
             Instant now = Instant.now();
 
-            if (now.isAfter(tokenExpiry.minus(WEB_REQUEST_TOKEN_EXPIRY_BUFFER_MINUTES, ChronoUnit.MINUTES))
+            if (tokenExpiry == OUTDATED_DATE || tokenRefreshDate == OUTDATED_DATE
+                    || now.isAfter(tokenExpiry.minus(WEB_REQUEST_TOKEN_EXPIRY_BUFFER_MINUTES, ChronoUnit.MINUTES))
```

Comment:
It seems like the bug was subtracting a value from `Instant.MIN`, which you have now worked around by subtracting a value from `Instant.EPOCH`? Instead, you could **add** a value instead, something like:
```java
now.plus(WEB_REQUEST_TOKEN_EXPIRY_BUFFER_MINUTES, ChronoUnit.MINUTES).isAfter(tokenExpiry)
```

?

Reply:
it should already be fixed by these additional checks
`tokenExpiry == OUTDATED_DATE || tokenRefreshDate == OUTDATED_DATE`
but just to be sure I will also rework the subtraction.

Reply:
Actually these additional checks are not needed anymore, since the problem of subtracting from `Instant.MIN` has now been resolved. So now just consider whether to simplify this by removing the checks. They rely on `OUTDATED_DATE` being used in all places, since you are comparing _references_, so it would not work for `tokenExpiry` being some other instance of `Instant.MIN`.

Reply:
I am in doubt that `tokenExpiry` will ever be set to another instance of `Instant.MIN` but you are right it could be simplified like this.